gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -400,6 +402,22 @@ public GsonBuilder setLenient() {
     return this;
   }
 
+  /**
+   * Enable default inclusion of type information and use this information during deserialization.
+   * Every complex type will be serialized/deserialized using two additional JSON fields:
+   * <ul>
+   *   <li>{@code _type}: The class name of the type as JSON string, e.g. {@code "my.CustomClass$Nested"}</li>
+   *   <li>{@code _properties}: The properties of the serialized instance in a JSON object.</li>
+   * </ul>
+   *
+   * @return a reference to this {@code GsonBuilder} object to fulfill the "Builder" pattern
+   */
+
+  public GsonBuilder enableDefaultTyping() {
```

Comment:
It would probably be good to have a big warning section here. I can try phrasing one if you like.

Reply:
Go ahead if you like